google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/scaladsl/GooglePubSub.scala

```diff
@@ -27,61 +28,46 @@ protected[pubsub] trait GooglePubSub {
   def publish(topic: String, config: PubSubConfig, parallelism: Int = 1)(
       implicit actorSystem: ActorSystem,
       materializer: Materializer
-  ): Flow[PublishRequest, immutable.Seq[String], NotUsed] = {
-    import materializer.executionContext
-
-    if (httpApi.isEmulated) {
-      Flow[PublishRequest].mapAsyncUnordered(parallelism) { request =>
-        httpApi.publish(config.projectId, topic, maybeAccessToken = None, request)
-      }
-    } else {
-      Flow[PublishRequest].mapAsyncUnordered(parallelism) { request =>
-        config.session.getToken().flatMap { accessToken =>
-          httpApi.publish(config.projectId, topic, Some(accessToken), request)
-        }
+  ): Flow[PublishRequest, immutable.Seq[String], NotUsed] =
+    httpApi
+      .accessToken[PublishRequest](config, parallelism)
+      .via(httpApi.publish(config.projectId, topic, parallelism))
+      .mapAsyncUnordered(parallelism) {
+        case (response, _) => response
       }
-    }
-  }
 
-  def subscribe(subscription: String, config: PubSubConfig)(
-      implicit actorSystem: ActorSystem
-  ): Source[ReceivedMessage, NotUsed] =
-    Source.fromGraph(
-      new GooglePubSubSource(
-        projectId = config.projectId,
-        session = config.session,
-        subscription = subscription,
-        returnImmediately = config.pullReturnImmediately,
-        maxMessages = config.pullMaxMessagesPerInternalBatch,
-        httpApi = httpApi
+  /**
+   * Creates a source pulling messages from subscription
+   */
+  def subscribe(subscription: String, config: PubSubConfig, parallelism: Int = 1)(
+      implicit actorSystem: ActorSystem,
+      materializer: Materializer
+  ): Source[ReceivedMessage, Cancellable] =
+    Source
+      .tick(0.seconds, 1.second, Done)
```

Comment:
This is nice. It might happen that the requests behave a bit differently depending on the downstream demand than with the graph stage, but I don't see any major difference.